File: DAOS/LayerDao/DaoADOdb.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.OleDb;

namespace DAOS.LayerDao
{
    public abstract class DaoADOdb
    {

        protected OleDbConnection cn; //mantém a conexão com o bd
        protected OleDbCommand cmd; // executa instruções de procedures
        protected OleDbDataReader reader; // consultas - modo conectado
        protected OleDbDataAdapter da; //Consultas modo desconectado

        private String conexao = ConfigurationManager.ConnectionStrings["MySQL"].ConnectionString;

        public DaoADOdb() {
            cn = new OleDbConnection(conexao);
            cmd = new OleDbCommand();
            da = new OleDbDataAdapter();
            
        }

        protected void AbrirConexao()
        {
            try
            {
                cn.Open();
                cmd.Connection = cn;
            }
            catch (Exception ex)
            {

                throw ex;
            }


        }

        protected void FecharConexao()
        {
            if (cn.State == System.Data.ConnectionState.Open)
            {

                cn.Close();
            }

        }

    }
}
}

File: GenericFunctions/LayerDao/DaoADOdb.vb
Imports System.Configuration
Imports System.Data
Imports System.Data.OleDb

Public Class DaoADOdb
    Protected cn As OleDbConnection
    Protected cmd As OleDbCommand
    Protected reader As OleDbDataReader
    Protected da As OleDbDataAdapter
    Protected conexao As String = ConfigurationManager.ConnectionStrings("DES").ConnectionString
    Sub DaoADOdb()
        Try
            cn = New OleDbConnection(conexao)
            cmd = New OleDbCommand
            da = New OleDbDataAdapter
        Catch ex As Exception
            Throw ex
        End Try
    End Sub

    Public Sub OpenConnection()
        Try
            cn.Open()
            cmd.Connection = cn
        Catch ex As Exception
            Throw ex
        End Try
    End Sub

    Public Sub ClosedConnection()
        If cn.State = ConnectionState.Open Then
            cn.Close()
        End If

    End Sub

End Class
